File: examples/simple/Cpp/reader.cpp
#include "adios2.h"
#include <mpi.h>
#include <thread>
#include <chrono>

int main(int argc, char **argv)
{
	MPI_Comm comm;
	MPI_Init(&argc, &argv);
	MPI_Comm_dup(MPI_COMM_WORLD, &comm);

    //@effis-init comm=comm
	adios2::ADIOS adios(comm);

    //@effis-begin "Jabberwocky"->"Jaberwocky"
	adios2::IO reader_io = adios.DeclareIO("Jabberwocky");
	adios2::Engine reader = reader_io.Open("Jabberwocky.bp", adios2::Mode::Read, comm);

    while (true)
	{
		adios2::StepStatus status = reader.BeginStep(adios2::StepMode::Read, 10.0);

        if (status == adios2::StepStatus::NotReady)
		{
            std::this_thread::sleep_for(std::chrono::milliseconds(1000));
            continue;
		}
		else if (status != adios2::StepStatus::OK)
		{
            break;
		}
		std::cout << reader.CurrentStep() << std::endl;
        reader.EndStep();
	}

    reader.Close();
    //@effis-end

   //@effis-finalize
}

File: examples/simple/Python/reader.py
#!/usr/bin/env python

import adios2
import time
import socket

class mpi:
    UseComm = True

if mpi.UseComm:
    from mpi4py import MPI


if __name__ == "__main__":
    print(socket.gethostname())

    if mpi.UseComm:
        comm = MPI.COMM_WORLD
        adios = adios2.ADIOS(comm)
    else:
        adios = adios2.ADIOS()
        comm = None

    #@effis-init comm=comm

    #@effis-begin "Jabberwocky"->"Jabberwocky"
    reader_io = ad.DeclareIO("Jabberwocky")
    #@effis-end

    if mpi.UseComm:
        #@effis-begin reader_io-->"Jabberwocky"
        reader = reader_io.Open("Jabberwocky.bp", adios2.Mode.Read, comm)
        #@effis-end
    else:
        #@effis-begin reader_io-->"Jabberwocky"
        reader = reader_io.Open("Jabberwocky.bp", adios2.Mode.Read)
        #@effis-end


    #@effis-begin reader--->"Jabberwocky"
    while True:

        status = reader.BeginStep(kittie.Kittie.ReadStepMode, 10.0)

        if status == adios2.StepStatus.NotReady:
            time.sleep(1)
            continue
        elif status != adios2.StepStatus.OK:
            break

        print(reader.CurrentStep())
        reader.EndStep()

    reader.Close()
    #@effis-end

    #@effis-finalize
